pkg/engine/mutation.go

```diff
@@ -52,7 +52,7 @@ func Mutate(policy kyverno.ClusterPolicy, resource unstructured.Unstructured) (r
 		if rule.Mutation.Overlay != nil {
 			var ruleResponse RuleResponse
 			ruleResponse, patchedResource = processOverlay(rule, resource)
-			if reflect.DeepEqual(ruleResponse, (RuleResponse{})) {
+			if strings.Contains(ruleResponse.Message, "policy not applied") {
```

Comment:
Can we identify if the policy is not applied by retuning errors from process overlay, instead of doing a string matching?